spring-cloud-gcp-starters/spring-cloud-gcp-starter-logging/src/main/java/org/springframework/cloud/gcp/logging/autoconfig/StackdriverLoggingAutoConfiguration.java

```diff
@@ -20,24 +20,64 @@
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.logging.CompositeTraceIdExtractor;
 import org.springframework.cloud.gcp.logging.LoggingWebMvcConfigurer;
+import org.springframework.cloud.gcp.logging.TraceIdExtractor;
 import org.springframework.cloud.gcp.logging.TraceIdLoggingWebMvcInterceptor;
+import org.springframework.cloud.gcp.logging.XCloudTraceIdExtractor;
+import org.springframework.cloud.gcp.logging.ZipkinTraceIdExtractor;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 /**
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  */
 @Configuration
 @ConditionalOnClass({ TraceIdLoggingWebMvcInterceptor.class, TraceLoggingEnhancer.class })
+@EnableConfigurationProperties({ StackdriverLoggingProperties.class })
 @ConditionalOnProperty(value = "spring.cloud.gcp.logging.enabled", matchIfMissing = true)
 @Import(LoggingWebMvcConfigurer.class)
 public class StackdriverLoggingAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public TraceIdLoggingWebMvcInterceptor getLoggingWebMvcInterceptor() {
-		return new TraceIdLoggingWebMvcInterceptor();
+	public TraceIdLoggingWebMvcInterceptor loggingWebMvcInterceptor(
+			TraceIdExtractor extractor) {
+		return new TraceIdLoggingWebMvcInterceptor(extractor);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public TraceIdExtractor traceIdExtractor(
+			StackdriverLoggingProperties loggingProperties) {
+		TraceIdExtractorType type = loggingProperties.getExtractorType();
+		TraceIdExtractorType checkedCombination = type == null
```

Comment:
You definitely don't need two variables. You can't override the previous value via ternary operator.
But nit-pick: it's not always a *combination*. Just *type* feels enough.

Reply:
Will change. I just didn't want to write `loggingProperties.getExtractorType()` twice.

Reply:
You don’t need call twice - just reuse the variable!